### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- SignIn Page:

Implemented the SignIn page utilizing reusable components for better modularity and code reuse.
Added an email input field with validation and error handling to enhance user experience.
Included a button to submit the user's credentials.
photo_2024-10-31_20-01-20
photo_2024-10-31_20-01-28

- LoginLogo Component:

Created the LoginLogo component to manage the application's logo display effectively.
Utilized props for flexibility, allowing for dynamic styling using the sx prop.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://codecraftersintership.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-3)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/SignIn/useSignIn.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    setEmail(event.target.value);
+  };
+
+  const handleSubmit = (): void => {
+    if (!email) {
+      setError('Email is required');
+      return;
+    }
+
+    console.log('Submitted email:', email);
```
Comment (lyudad) [inline src/pages/SignIn/useSignIn.ts]: 1.1.8

File: src/pages/SignIn/useSignIn.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  const handleEmailChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    setEmail(event.target.value);
+  };
+
+  const handleSubmit = (): void => {
+    if (!email) {
+      setError('Email is required');
```
Comment (lyudad) [inline src/pages/SignIn/useSignIn.ts]: 1.3.3

File: src/pages/components/LoginLogo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    mainImageAlt: t('altText.logo'),
+    topLeftImage: {
+      alt: t('altText.key'),
+      sx: { top: '-13px', left: '10px' },
+    },
+    bottomRightImage: {
+      alt: t('altText.lock'),
+      sx: { bottom: '-10px', right: '0px' },
+    },
+  };
```
Comment (lyudad) [inline src/pages/components/LoginLogo/index.tsx:27]: move it to stypes file

File: src/pages/SignIn/useSignIn.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    emailInput: {
+      label: t('signin.email'),
+      type: 'email' as const,
+    },
+    submitButton: {
+      label: t('signin.submit'),
+      variant: 'colored' as const,
+    },
+  };
```
Comment (lyudad) [inline src/pages/SignIn/useSignIn.ts]: тут какая то магия, что ты пытался сделать этой переменной?

Comment (lyudad) [reply src/pages/SignIn/useSignIn.ts]: сразу в компоненте переводи, не делай такие страшные обьекты. перемудрил

File: src/pages/SignIn/useSignIn.ts (hunk 1 above)
Comment (oyakovytskyi) [reply src/pages/SignIn/useSignIn.ts]: my bad, removed

File: src/pages/SignIn/useSignIn.ts (hunk 2 above)
Comment (oyakovytskyi) [reply src/pages/SignIn/useSignIn.ts]: changed

File: src/pages/components/LoginLogo/index.tsx (hunk 3 above)
Comment (oyakovytskyi) [reply src/pages/components/LoginLogo/index.tsx:27]: done

File: src/pages/SignIn/useSignIn.ts (hunk 4 above)
Comment (oyakovytskyi) [reply src/pages/SignIn/useSignIn.ts]: remove whole const, just placed text straight with i18n

**Review by lyudad — APPROVED** [review]

**Pull request merged**